src/ranked_match.cc

```diff
@@ -36,9 +36,11 @@ bool matches(UsedLetters query, UsedLetters letters)
 
 using Utf8It = utf8::iterator<const char*>;
 
-static int count_word_boundaries_match(StringView candidate, StringView query)
+static int count_word_boundaries_match_first(StringView candidate, StringView query, int &first_word_len)
```

Comment:
declare a struct WordBoundaries { int count; int first_word_len; } and return that, instead of an out parameter.